LSpec/LSpec.lean

```diff
@@ -5,7 +5,8 @@ import LSpec.SlimCheck.Checkable
 
 # The core `LSpec` framework
 
-## TODO: Add all relavent documentation
+## Add all relavent documentation
+Check [here](./LSpec/Testing.lean) for examples of uses 
```

Comment:
That testing file is just for drafts to make sure that everything works. Maybe we should have a proper file with examples instead.

Reply:
I reverted the changes to `Testing.lean` and created a separate `Examples.lean`.